### PR TITLE
Update Ruby version, Dockerfile, and Gemfile

### PR DESCRIPTION
**Pull request by k2nr**
- Ruby version 2.6
- Rewrote Dockerfile to use https://github.com/degica/dockerfiles/blob/master/rails-buildpack/2.6/Dockerfile and multi-stage build
- Install per-service aws-sdk gems instead of the all-in-one `aws-sdk`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+      openssh-client \
+ && rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /usr/local/bundle /usr/local/bundle
+ADD --chown=app:app . $APP_HOME
 
-ADD . $APP_HOME
-RUN chown -R app:app $APP_HOME
 USER app
-RUN rake assets:precompile
```
Comment (essa) [inline Dockerfile]: Where and when are assets compiled?

Comment (k2nr) [reply Dockerfile]: Well I've noticed Barcelona is API only and doesn't have any assets :sweat_smile: so I just removed this line

Comment (essa) [reply Dockerfile]: Ah I see

**Comment by essa** [comment]
LGTM 👍

**Comment by k2nr** [comment]
Thanks!

**Pull request merged**